client/extensions/woocommerce/woocommerce-services/views/shipping-label/label-purchase-modal/purchase-button.js

```diff
@@ -0,0 +1,107 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import React from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { bindActionCreators } from 'redux';
+import { localize } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import Button from 'components/button';
+import getPDFSupport from 'woocommerce/woocommerce-services/lib/utils/pdf-support';
+import formatCurrency from 'lib/format-currency';
+import {
+	confirmPrintLabel,
+	purchaseLabel,
+} from 'woocommerce/woocommerce-services/state/shipping-label/actions';
+import {
+	getShippingLabel,
+	isLoaded,
+	getTotalPriceBreakdown,
+	canPurchase,
+} from 'woocommerce/woocommerce-services/state/shipping-label/selectors';
+
+class PurchaseButton extends React.Component {
```

Comment:
Why is this a `React.Component` instead of a "function" component? Personal preference, or is it preferred now?

Reply:
I wasn't originally going after refactoring that wasn't module-related, but I'm more than happy to change it to a function.